ExternalProjects/BizHawk.SrcGen.VersionInfo/VersionInfoGenerator.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+
+namespace BizHawk.SrcGen.VersionInfo;
+
+[Generator]
+public class VersionInfoGenerator : ISourceGenerator
+{
+	public void Initialize(GeneratorInitializationContext context)
+	{
+	}
+
+	private static string? ExecuteGitWithArguments(string arguments)
+	{
+		var startInfo = new ProcessStartInfo("git", arguments)
+		{
+			RedirectStandardOutput = true
+		};
+		try
+		{
+			using Process git = Process.Start(startInfo) ?? throw new Exception("Failed to start git process");
+			git.WaitForExit();
+			return git.StandardOutput.ReadLine();
+		}
+		catch (Exception e)
+		{
+			return $"{e.GetType()}: {e.Message}";
+		}
+	}
+
+	public void Execute(GeneratorExecutionContext context)
+	{
+		// Finds the current project directory in order to pass to git commands.
+		// This is written in a way to (hopefully) work both for build and IDE analyzers
+		// FIXME: This should probably be done in a better way, but I haven't found any
+		string projectDir = Path.GetDirectoryName(context.Compilation.SyntaxTrees.First(x => x.HasCompilationUnitRoot && x.FilePath.Contains("BizHawk.Common")).FilePath)!;
```

Comment:
How does this not fail on the rest of the projects? `PrivateAssets="all"` from my other suggestion would fix it in any case.

Reply:
I can actually not find any documentation on this, but to me it appears like `PrivateAssets="all"` is doing the same to a `ProjectReference` as `Analyzer` is already doing implicitly? In any case I could not find any behavioral difference between the two in my limited testing.

Reply:
Go ahead and check-in the assembly then.

Reply:
> Go ahead and check-in the assembly then.

Not sure what you mean, the assembly is committed and should be working fine.